neural_speed/models/whisper/whisper.cpp

```diff
@@ -2434,21 +2434,14 @@ void whisper_free(struct whisper_context* ctx) {
     if (ctx->model.ctx) {
       ne_free(ctx->model.ctx);
     }
-    if (ctx->model.buf) {
-      delete ctx->model.buf;
-    }
+    delete ctx->model.buf;
```

Comment:
why remove the condition? how about using std-way to manage buffer?

Reply:
@a32543254 will linjun work on re-implementing whisper with more modern coding?

Reply:
> why remove the condition?

[readability-delete-null-pointer] shall I remove it?

Reply:
`The check is unnecessary as deleting a null pointer has no effect.` Here is clang-tidy's explanation. I believe it's different for GCC and MSVC.

Reply:
fixed in https://github.com/intel/neural-speed/pull/29/commits/0dc61cd53afb5c37f3a13897108ef5cf69e313ed